nrf52-hal-common/src/twim.rs

```diff
@@ -230,6 +233,114 @@ impl<T> Twim<T> where T: TwimExt {
             return Err(Error::Receive);
         }
 
+        // Conservative compiler fence to prevent optimizations that do not
+        // take in to account DMA
+        compiler_fence(AcqRel);
```

Comment:
The comments I made in `spim.rs` about a (possibly) missing compiler fence at the method entry and the early return apply here too.